src/native/mqtt_connection.c

```diff
@@ -434,17 +498,10 @@ static void s_on_shutdown_disconnect_complete(struct aws_mqtt_client_connection
         return;
     }
 
-    jobject mqtt_connection = (*env)->NewLocalRef(env, jni_connection->java_mqtt_connection);
-    if (mqtt_connection != NULL) {
-        (*env)->CallVoidMethod(env, mqtt_connection, crt_resource_properties.release_references);
-
-        (*env)->DeleteLocalRef(env, mqtt_connection);
-
-        aws_jni_check_and_clear_exception(env);
-    }
+    /* Release the underlying mqtt connection */
+    aws_mqtt_client_connection_release(jni_connection->client_connection);
```

Comment:
With these changes, we shouldn't even need to acquire the env anymore.  This function should just be a aws_mqtt_client_connection_release call.